src/utils/async.js

```diff
@@ -0,0 +1,19 @@
+export const timeout = (func, onTimeout = () => {}, timeoutMs = 10000) =>
+  Promise.race([
+    func,
+    typeof timeoutMs === 'number' &&
+      new Promise(resolve => setTimeout(resolve, timeoutMs)).then(onTimeout)
+  ]);
+
+export const tryUntilSuccessful = async (func, maxRetries = 1000, timeoutMs = 1000) => {
```

Comment:
We should probably set the max retries to something a lot lower, and show a screen that says something like 'Could not load data. Please click to retry' when the retries are up